src/io/syscall.cr

```diff
@@ -62,22 +62,26 @@ module IO::Syscall
   end
 
   def write_syscall_helper(slice : Bytes, errno_msg : String) : Nil
-    loop do
-      bytes_written = yield slice
-      if bytes_written != -1
-        slice += bytes_written
-        return if slice.size == 0
-      else
-        if Errno.value == Errno::EAGAIN
-          wait_writable
+    return if slice.empty?
+
+    begin
```

Comment:
Why extra `begin ... end`?

Reply:
Because the `ensure` must not run if the slice is empty. It's pointless to do anything at all if the slice is empty.

Reply:
Yeah, I meant that `ensure` could be attached to the `loop` instead, like it was before.

Reply:
It was attached to the `def`